### PR TITLE
DEP: drop support for Python 3.9, require 3.10 or newer

### PR DESCRIPTION
**Pull request by neutrinoceros**
Nothing urgent, but as I'm currently spending lots of time waiting for wheels to build, I figure it may be time to remove these.

For reference, the previous such PR, #2471, was merged almost a year ago.

### CODE REVIEW

**Review by tacaswell — APPROVED** [review]

**Comment by tacaswell** [comment]
Something in not right, azure did not fire on this PR.

**Comment by neutrinoceros** [comment]
Is there a way to retrigger it, other than a power-cycle ?

**Comment by tacaswell** [comment]
> Is there a way to retrigger it, other than a power-cycle ?

for individual jobs there are buttons in the UI, but given we missed a whole CI service, just power-cycled.

**Comment by neutrinoceros** [comment]
> AppVeyor was unable to build non-mergeable pull request

Hum, weird, I see no conflicts here. Let's try a rebase still

**Comment by tacaswell** [comment]
This makes we worried the changes broke the azure configuration.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -129,9 +129,9 @@ jobs:
...
       # -mindeps : test with oldest supported version of (python) dependencies
       # -deps-pre : test pre-release versions of (python) dependencies)
-      py39:
-        python.version: '3.9'
-        TOXENV: py39-test-deps,py39-test-mindeps,py39-test-deps-pre
+      py310:
+        python.version: '3.10'
+        TOXENV: py310-test-deps,py310-test-mindeps,py310-test-deps-pre
         H5PY_ENFORCE_COVERAGE: yes
       py310:
```
Comment (takluyver) [inline azure-pipelines.yml]: We've got a duplicate key here, that might be what's breaking Azure.

```suggestion
      py310:
```

File: azure-pipelines.yml [hunk 2]
```diff
@@ -87,18 +87,18 @@ jobs:
     # -deps-pre : test pre-release versions of (python) dependencies)
     #
     # 64 bit - HDF5 1.10
-      py39-hdf5110:
-        python.version: '3.9'
-        TOXENV: py39-test-deps,py39-test-mindeps,py39-test-deps-pre
+      py310-hdf5110:
```
Comment (takluyver) [inline azure-pipelines.yml:90]: We were roughly taking a diagonal across the matrix of possible dependencies - testing a few different Python versions and HDF5 versions at the same time. Now they're getting all bunched up on the same Python version. We might want to shuffle some along to newer Python versions - or we could decide it's redundant with the wheel builds on Github actions and cut some of them out. :shrug:

Comment (neutrinoceros) [reply azure-pipelines.yml:90]: your call ! I wasn't aware this was the strategy, but I think I'd be comfortable with dropping some of these.

Comment (takluyver) [reply azure-pipelines.yml:90]: I'm comfortable dropping some at this point - I think we've got pretty good coverage by building and testing wheels on GHA. There are a couple of corner cases it would be good to keep, though, like the C locale tests and the MPI tests. The Mac tests here also install HDF5 from Homebrew, whereas on GHA we directly build it from source, so it's probably worth keeping one or two jobs doing that.

**Comment by takluyver** [comment]
And I agree that we can reasonably drop Python 3.9 now. We'll be adding 3.14 before long.

**Comment by neutrinoceros** [comment]
Actually I already added support for 3.14, the only things we're missing are
- stable wheels (next release will have them)
- non-nightly tests, which require 3.14.0 to be available on GHA

**Comment by neutrinoceros** [comment]
Ok I think this one is go now. Thanks all !

**Comment by takluyver** [comment]
Thanks, nice one!
